components/client/src/test/unit/java/com/hotels/styx/client/connectionpool/SimpleConnectionPoolTest.java

```diff
@@ -543,10 +547,54 @@ public void countsTerminatedConnections() throws Exception {
         assertThat("busy connections ", this.connectionPool.stats().busyConnectionCount(), is(0));
     }
 
+    @Test
+    public void verifiesMaxUsageTerminationIsOffWithNegativeSetting() {
+        this.connectionPool = connectionPool(
+                connectionPoolConfig()
+                        .maxConnectionsPerHost(1)
+                        .connectionExpirationSeconds(-1, SECONDS));
+        Connection.Listener listener = mock(Connection.Listener.class);
+        Connection listenedConnection = borrowConnectionSynchronously();
+        listenedConnection.addConnectionListener(listener);
+        connectionPool.returnConnection(listenedConnection);
+
+        for (int i = 0; i < 1000000; i++ ) {
+            Connection connection = borrowConnectionSynchronously();
+            assertThat(connection, is(listenedConnection));
+            connectionPool.returnConnection(connection);
+        }
+
+        verifyZeroInteractions(listener);
+    }
+
+    @Test
+    public void verifiesMaxUsageTerminationIsOn() {
+        this.connectionPool = connectionPool(connectionPoolConfig().connectionExpirationSeconds(1, SECONDS));
+
+        Connection.Listener listener = mock(Connection.Listener.class);
+
+        Connection connectionExpired = borrowConnectionSynchronously();
+        connectionExpired.addConnectionListener(listener);
+        connectionPool.returnConnection(connectionExpired);
+
+        try {
+            Thread.sleep(SECONDS.toMillis(1));
+        } catch (InterruptedException e) {
+            fail();
```

Comment:
Why not just throw `InterruptedException`?

Reply:
Also, why are we using a delay here? If this behaviour depends on the passage of time, we have a mockable interface called `com.hotels.styx.api.Clock` that can be used for this. (There is a real implementation in `com.hotels.styx.api.Clocks`).